idaes/core/base/tests/test_extended_control_volume_0d.py

```diff
@@ -0,0 +1,167 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2024 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+Tests for ExtendedControlVolumeBlockData.
+
+Author: Andrew Lee
+"""
+import pytest
+from pyomo.environ import ConcreteModel, Constraint, units
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import (
+    ExtendedControlVolume0DBlock,
+    FlowsheetBlockData,
+    declare_process_block_class,
+)
+from idaes.core.util.exceptions import (
+    ConfigurationError,
+)
+from idaes.core.util.testing import (
+    PhysicalParameterTestBlock,
+)
+
+
+# -----------------------------------------------------------------------------
+# Mockup classes for testing
+@declare_process_block_class("Flowsheet")
+class _Flowsheet(FlowsheetBlockData):
+    def build(self):
+        super(_Flowsheet, self).build()
+
+
+@pytest.mark.unit
+def test_add_isothermal_constraint():
+    m = ConcreteModel()
+    m.fs = Flowsheet(dynamic=False)
+    m.fs.pp = PhysicalParameterTestBlock()
+
+    m.fs.cv = ExtendedControlVolume0DBlock(property_package=m.fs.pp)
+
+    m.fs.cv.add_geometry()
+    m.fs.cv.add_state_blocks(has_phase_equilibrium=False)
+
+    m.fs.cv.add_isothermal_constraint()
+
+    assert isinstance(m.fs.cv.enthalpy_balances, Constraint)
+    assert len(m.fs.cv.enthalpy_balances) == 1
+    assert str(m.fs.cv.enthalpy_balances[0].expr) == str(
+        m.fs.cv.properties_in[0].temperature == m.fs.cv.properties_out[0].temperature
+    )
+
+    assert_units_consistent(m.fs.cv)
+
+
+@pytest.mark.unit
+def test_add_isothermal_constraint_dynamic():
+    m = ConcreteModel()
+    m.fs = Flowsheet(dynamic=True, time_set=[0, 1, 2, 3], time_units=units.s)
+    m.fs.pp = PhysicalParameterTestBlock()
+
+    m.fs.cv = ExtendedControlVolume0DBlock(property_package=m.fs.pp)
+
+    m.fs.cv.add_geometry()
+    m.fs.cv.add_state_blocks(has_phase_equilibrium=False)
+
+    m.fs.cv.add_isothermal_constraint()
+
+    assert isinstance(m.fs.cv.enthalpy_balances, Constraint)
+    assert len(m.fs.cv.enthalpy_balances) == 4
+    for t in m.fs.time:
+        assert str(m.fs.cv.enthalpy_balances[t].expr) == str(
+            m.fs.cv.properties_in[t].temperature
+            == m.fs.cv.properties_out[t].temperature
+        )
+
+    assert_units_consistent(m.fs.cv)
```

Comment:
Interesting- I thought I remember the units check failing for dynamic cases, or is that only when has_holdup=True as well?

Reply:
It will for anything with a partial derivative. Here we get aways with it because we don't have a derivative.

Reply:
Not necessarily. Generally for spatial domains we normalize the `ContinuousSet` to run from 0 to 1, adding in the flow length later. Those should pass. Any `ContinuousSet` with dimensions, however, will cause it to fail.